enterprise/api/permissions.py

```diff
@@ -0,0 +1,30 @@
+"""
+Permission classes for the Enterprise API.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from rest_framework import permissions
+
+from enterprise.api.utils import get_service_usernames
+
+
+class IsServiceUserOrReadOnly(permissions.IsAuthenticated):
+    """
+    This request is authenticated as a service user, or is a read-only request.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Enforce that the user is authenticated. If the request isn't read-only (unsafe), then deny permission unless
+        the user is a service user.
+        """
+        authenticated = super(IsServiceUserOrReadOnly, self).has_permission(request, view)
+
+        service_users = get_service_usernames()
+
+        return (
```

Comment:
I know that this is a perfectly acceptable way to write code, but imho it's much less readable than doing the work prior to the return and then returning a simple answer.  In addition to readability I find it harder to write tests for implementations like this as well.  It's just MHO, but wanted to toss it out there.